common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -623,8 +641,26 @@ function (Sjson, AsyncProcess) {
         play: function () {
             if (this.loaded) {
                 if (!this.rendered) {
-                    var start = this.sjson.getStartTimes(),
-                        captions = this.sjson.getCaptions();
+                    // get start and caption. If startTime and endTime
+                    // are specified, filter by that range.
+                    var start,
+                        captions,
+                        startTime, 
+                        endTime,
+                        results,
+                        startTimeInSec = state.config.startTime,
+                        endTimeInSec = state.config.endTime;
+
+                    if (endTimeInSec !== null) {
```

Comment:
Yep, the logic is mostly identical: find start/end time, then find the starts and captions that fall between this range. Refactoring into a function `getBoundedCaptions` for upcoming commit.